### PR TITLE
Use a document-centric processing model.

### PR DESCRIPTION
**Pull request by szager-chromium**
This should integrate better with the existing html event loop spec.

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -468,12 +472,12 @@ the <a>Intersection Observer</a> API.
...
-<h4 id='browsing-contexts-defines'>
-Browsing Contexts</h4>
+<h4 id='document-defines'>
+Document</h4>
 
-Each <a>unit of related similar-origin browsing contexts</a> has an
-<dfn for="browsing context">IntersectionObserverTaskQueued</dfn> flag which
-is initialized to false.
+Each <a>Document</a> has an
+<dfn for="browsing context">IntersectionObserverTaskQueued</dfn> flag
```
Comment (mpb) [inline index.bs]: Is \<dfn for="browsing context"> still accurate?

File: index.bs [hunk 2]
```diff
@@ -543,15 +547,16 @@ similar-origin browsing contexts</a> |unit|, run these steps:
 <h4 id='queue-intersection-observer-entry-algo'>
 Queue an IntersectionObserverEntry</h4>
 
-To <dfn>queue an IntersectionObserverEntry</dfn> for |observer|, given a
-<a>unit of related similar-origin browsing contexts</a> |unit|, |observer|,
-|time|, |rootBounds|, |boundingClientRect|, |intersectionRect| and |target|,
+To <dfn>queue an IntersectionObserverEntry</dfn> for an {{IntersectionObserver}} |observer|, given a
+<a>Document</a> |document|; {{DOMHighResTimeStamp}} |time|;
+{{DOMRect}}'s |rootBounds|, |boundingClientRect|, and |intersectionRect|;
```
Comment (mpb) [inline index.bs]: Why is DOMRect's possessive?

**Comment by mpb** [comment]
Looks good to me, modulo some tiny style nits.

File: index.bs (hunk 1 above)
Comment (szager-chromium) [inline index.bs]: Changed to <dfn for="Document">

File: index.bs (hunk 2 above)
Comment (szager-chromium) [inline index.bs]: Removed the apostrophe.  English is hard.

**Comment by domenic** [comment]
Interesting. This changes not only the "run the update intersection observations steps" to be document-centric, but the whole spec. This now mismatches mutation observers---but maybe that's OK?

**Comment by szager-chromium** [comment]
@domenic I didn't refer to the MutationObserver spec when making this change, but I think the whole thing reads better when it only refers to documents.  WDYT?

**Comment by domenic** [comment]
It's hard for me to say. Maybe @ajklein or @annevk know why mutation observer is in terms of event loops/units of related similar-origin browsing contexts, instead of in terms of documents?

It seems quite possible that the mutation observer design is not appropriate for intersection observers, and documents is better, but it's worth checking first.

**Comment by ajklein** [comment]
@annevk probably knows more precisely, but MutationObserver, implementation-wise, is in terms of "changes that have occurred on this turn on this thread", which happens to map to "similar-origin browsing contexts".

**Comment by annevk** [comment]
@domenic mutation observers is this way because mutations can involve multiple documents (e.g., when you append a node that is from another document).

**Pull request merged**